src/inc/msquic.h

```diff
@@ -964,6 +964,10 @@ typedef struct QUIC_LISTENER_EVENT {
         struct {
             const QUIC_NEW_CONNECTION_INFO* Info;
             HQUIC Connection;
+            _Field_range_(0, QUIC_MAX_ALPN_LENGTH)
+            uint8_t NewNegotiatedAlpnLength;
+            _Field_size_bytes_(NewNegotiatedAlpnLength)
+            const uint8_t* NewNegotiatedAlpn;
```

Comment:
This must point into the `ClientAlpnList` in the `QUIC_NEW_CONNECTION_INFO`, which is in TLS format, where (I believe) it's 1-byte length prefixed. So then you don't need `NewNegotiatedAlpnLength`.

Reply:
So, instead of allocating a new heap area for this, I have to find the given alpn from the list and just point it directly?

Reply:
I've changed it in this way, thanks for helping on Discord!